crates/forge/bin/cmd/test/mod.rs

```diff
@@ -227,69 +227,71 @@ impl TestArgs {
         let outcome = self
             .run_tests(runner, config.clone(), verbosity, filter.clone(), test_options.clone())
             .await?;
-        let tests = outcome.clone().into_tests();
 
-        let mut decoded_traces = Vec::new();
-        let mut decoders = Vec::new();
-        for test in tests {
```

Comment:
It seemed that this section of decoder building was only relevant if `should_debug` is true. Because of this, if we require that `should_debug` is true before looping through each test and building those decoders, we can skip that unnecessary computation.